nixops/known_hosts.py

```diff
@@ -53,20 +53,18 @@ def rewrite(l):
             os.rename(tmp, path)
 
 
-def remove(ip_address, public_host_key):
+def remove(ip_address: str, public_host_key: str) -> None:
     """Remove a specific known host key."""
     _rewrite(ip_address, False, public_host_key)
 
 
-def add(ip_address, public_host_key):
+def add(ip_address: str, public_host_key: str) -> None:
     """Add a known host key."""
     _rewrite(ip_address, True, public_host_key)
 
 
-def update(prev_address, new_address, public_host_key):
-    assert public_host_key is not None
+def update(prev_address: str, new_address: str, public_host_key: str) -> None:
     # FIXME: this rewrites known_hosts twice.
-    if prev_address is not None and prev_address != new_address:
+    if prev_address != new_address:
         remove(prev_address, public_host_key)
-    if new_address is not None:
-        add(new_address, public_host_key)
+    add(new_address, public_host_key)
```

Comment:
As a heads up, this is a backwards incompatible API change to the function.  I think it's a good change, but previously plugins like [nixops-aws were calling "update" with None values to mean "remove" or "add"](https://github.com/NixOS/nixops-aws/pull/38).  I've fixed nixops-aws by having it call `add` or `remove` directly, but it may be worth scanning other plugins to see how they're calling `update`.